src/tensor_conversion.cpp

```diff
@@ -221,6 +221,23 @@ static Status validateTensor(const std::shared_ptr<const TensorInfo>& tensorInfo
     return StatusCode::OK;
 }
 
+inline static int getNumberOfInputs(const std::string* buffer) {
+    int numberOfInputs = 0;
+    size_t offset = 0;
+    if (buffer->size() < sizeof(uint32_t)) {
```

Comment:
This should also be checked inside while loop, right?
Example:

[4, 0, 0, 0, a, b, c, d, 2, 0]
would crash in line 231

If so, please add test for that if possible

Reply:
The same goes for L368-373

Reply:
And convertStringRequestFromBufferToOVTensor2D